app/code/Magento/InventoryLowQuantityNotification/Model/ResourceModel/BulkConfigurationTransfer.php

```diff
@@ -65,37 +65,35 @@ public function execute(
 
         $types = $this->getProductTypesBySkus->execute($skus);
 
-        foreach ($types as $sku => $type) {
-            if ($this->isSourceItemManagementAllowedForProductType->execute($type)) {
-                foreach ($skus as $sku) {
-                    $qry = $connection
-                        ->select()
-                        ->from($tableName, 'notify_stock_qty')
-                        ->where('sku = ?', $sku)
-                        ->where('source_code = ?', $originSource);
+        foreach ($skus as $sku) {
+            if ($this->isSourceItemManagementAllowedForProductType->execute($types[$sku])) {
+                $qry = $connection
+                    ->select()
+                    ->from($tableName, 'notify_stock_qty')
+                    ->where('sku = ?', $sku)
+                    ->where('source_code = ?', $originSource);
 
-                    $res = $connection->fetchOne($qry);
+                $res = $connection->fetchOne($qry);
 
-                    $notifyStockQty = ($res === null || $res === false) ? null : (float) $res;
-                    try {
-                        $connection->insert(
+                $notifyStockQty = ($res === null || $res === false) ? null : (float) $res;
+                try {
+                    $connection->insert(
+                        $tableName,
+                        [
+                            'source_code' => $destinationSource,
+                            'sku' => $sku,
+                            'notify_stock_qty' => $notifyStockQty,
+                        ]
+                    );
```

Comment:
it's better not insert DB records one be one, but accumulate all of them on PHP level and insert them all together with 1 SQL query INSERT on duplicate UPDATE (not updating with Source Code and SKU exists)